### PR TITLE
(2123) Users can export budget data

### PR DESCRIPTION
**Pull request by pezholio**
This allows users to export data about budgets. BEIS users can export budgets for all activities, or specific organisations, while DPs can export data for their own organisation.

# Screenshots

## Export page

image

## Organisation export page (BEIS user)

image

## Organisation export page (Delivery partner)

image

### CODE REVIEW

**Comment by pezholio** [comment]
[Users can export budget data](https://trello.com/c/SNpQakMp/2123-users-can-export-budget-data)

**Comment by mec** [comment]
@pezholio this is totally my fault: could we add the Delivery partner organisation name to the columns in the file? Ideally only on the one that exports all delivery partner data, but if it's in both it is not a problem. It can go in after `Delivery partner identifier` (column C).

This is so a BEIS user can sort and filter the data by delivery partner without relying on the RODA identifier, which can be problematic.

File: app/controllers/staff/exports/organisations_controller.rb [hunk 1]
```diff
@@ -24,7 +24,7 @@ def actuals
     respond_to do |format|
       format.csv do
         activities = Activity.where(organisation: @organisation)
-        export = QuarterlyActualExport.new(activities)
+        export = Actual::Export.new(activities)
```
Comment (mec) [inline app/controllers/staff/exports/organisations_controller.rb:27]: Thank you for doing this - a great start to a PR - make thing better! 👍

**Review by mec — CHANGES_REQUESTED** [review]
I've looked this over and poked it in the app - it is shapping up super nicely.

I do feel like there is something amiss though, I was expecting to see:

- multiple rows where an activity has multiple budgets for the same financial year
- level B (programme) budgets should be in the org specific file, I think this is our old friend the many organisation roles again

Happy to have chat about this if helpful, just let me know?

File: app/models/financial_year.rb [hunk 2]
```diff
@@ -63,4 +63,12 @@ def to_s
   def quarters
     FinancialQuarter::QUARTERS.map { |q| FinancialQuarter.new(start_year, q) }
   end
+
+  def pred
```
[the remark this answers is not in the document]
Comment (pezholio) [reply app/models/financial_year.rb:67]: This is an internal Ruby thing, that makes the class iterable. There's a test in the FinancialYear spec

[the remark this answers is not in the document]
Comment (pezholio) [reply app/models/financial_year.rb:67]: I'll edit the commit message - I agree though, `pred` does sound better!

**Review by mec — APPROVED** [review]

**Pull request merged**